glean-core/python/glean/net/ping_upload_worker.py

```diff
@@ -75,65 +71,78 @@ def _test_process_sync(cls) -> bool:
 )
 
 
-def _process(storage_dir: Path, configuration) -> bool:
-    success = True
+def _parse_ping_headers(
+    headers_as_json: str, document_id: str
+) -> List[Tuple[str, str]]:
+    """
+    Parse the headers coming from FFI.
 
-    log.debug("Processing persisted pings at {}".format(storage_dir.resolve()))
+    Args:
+        headers_as_json (str): The JSON key-value map of the headers.
+        document_id (str): The id of the document the headers are for.
 
-    try:
-        for path in storage_dir.iterdir():
-            if path.is_file():
-                if _FILE_PATTERN.match(path.name):
-                    log.debug("Processing ping: {}".format(path.name))
-                    if not _process_file(path, configuration):
-                        log.error("Error processing ping file: {}".format(path.name))
-                        success = False
-                else:
-                    log.debug("Pattern mismatch. Deleting {}".format(path.name))
-                    path.unlink()
-    except FileNotFoundError:
-        log.debug("File not found: {}".format(storage_dir.resolve()))
-        success = False
-
-    return success
-
-
-def _process_file(path: Path, configuration) -> bool:
+    Returns:
+        headers (list of (str, str)): The headers to send.
     """
-    Processes a single ping file.
-    """
-    processed = False
-
+    headers = []
     try:
-        with path.open("r", encoding="utf-8") as fd:
-            lines = iter(fd)
-            try:
-                url_path = next(lines).strip()
-                serialized_ping = next(lines)
-                valid_content = True
-            except StopIteration:
-                valid_content = False
-        # On Windows, we must close the file before deleting it
-        if not valid_content:
-            path.unlink()
-            log.error("Invalid ping content in {}".format(path.resolve()))
-            return False
-    except FileNotFoundError:
-        log.error("Could not find ping file {}".format(path.resolve()))
-        return False
-    except IOError as e:
-        log.error("IOError when reading {}: {}".format(path.resolve(), e))
-        return False
-
-    processed = configuration.ping_uploader.do_upload(
-        url_path, serialized_ping, configuration
-    )
-
-    if processed:
-        path.unlink()
-        log.debug("{} was deleted".format(path.name))
-
-    return processed
+        parsed_headers = json.loads(headers_as_json)
+        for header_name, header_value in parsed_headers.items():
+            headers.append((header_name, header_value))
```

Comment:
Nit: I think the above 4 lines can be replaced with `headers = list(json.loads(headers_as_json).items())`